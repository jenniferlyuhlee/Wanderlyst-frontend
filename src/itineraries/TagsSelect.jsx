import React, { useContext } from "react";
import UserContext from "../auth-user/UserContext";

function TagsSelect( {selectedTags, handleTagChange} ){
    const { tags } = useContext(UserContext);

    return(
        <div className = "Form-tag-check row mt-2">
            <h5 className = "mb-1 p-0">Tags:</h5>
            {tags.map(tag => (
            <div key={tag.id} className="form-check">
                <input className="form-check-input col" 
                    type="checkbox" 
                    value={tag.id} 
                    id={tag.name}
                    onChange = {handleTagChange}
                    checked = {selectedTags.includes(String(tag.id))}
                />
                <label className="form-check-label" htmlFor={tag.name}>
                    {tag.name}
                </label>
            </div>
            ))}
        </div>
    );
}

export default TagsSelect;