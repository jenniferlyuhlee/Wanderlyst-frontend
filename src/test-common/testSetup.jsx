import React, { useState } from "react";
import UserContext from "../auth-user/UserContext";
import WanderlystApi from "../utils/api";
import '@testing-library/jest-dom';

const demoUser = {
  username: "testadmin", 
  firstName: "testfirst",
  lastName: "testlast",
  location: "testloc",
  bio: "testbio",
  profilePic: null,
  isAdmin: true,
};

const demoTags = [
  {id: 1, name: 'Foodie', description: 'Savor the world one bite at a time. Dive into a journey packed with delicious local cuisines, hidden culinary gems, and must-try dishes that will leave your taste buds tingling.'},
  {id: 2, name: 'Explorer', description: 'Step off the beaten path and uncover the unknown. From ancient ruins to vibrant cityscapes, let your curiosity guide you through unforgettable experiences.'}
]

const UserProvider = ({ children, 
                        currUser = demoUser, 
                        setCurrUser, 
                        tags = demoTags,
                        logout,
                        hasLikedItin,
                        toggleLike }) => {
  const [user, setUser] = useState(currUser);

  return (
    <UserContext.Provider value={{ currUser: user, 
                                  setCurrUser: setUser, 
                                  tags, logout, 
                                  hasLikedItin, toggleLike}}>
      {children}
    </UserContext.Provider>
  );
};

export { UserProvider };