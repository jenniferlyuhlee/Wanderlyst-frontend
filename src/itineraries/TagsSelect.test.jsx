import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import { UserProvider } from "../test-common/testSetup";
import { MemoryRouter} from "react-router-dom";
import TagsSelect from "./TagsSelect";
import { expect, describe, test} from "vitest";


describe("TagsSelect sub-form component", () => {
    const mockSelectedTags = ["1"];
    const mockHandleTagChange = vi.fn();

    test("renders without crashing", () => {
        render(
            <MemoryRouter>
                <UserProvider>
                    <TagsSelect selectedTags={mockSelectedTags}
                                handleTagChange={mockHandleTagChange}/>
                </UserProvider>
            </MemoryRouter>)
    });

    test("matches snapshot", () => {
        const {asFragment} = render(
            <MemoryRouter>
                <UserProvider>
                    <TagsSelect selectedTags={mockSelectedTags}
                                handleTagChange={mockHandleTagChange}/>
                </UserProvider>
            </MemoryRouter>);
        expect(asFragment()).toMatchSnapshot();
    });

    test("handles selecting tags", () => {
        const {getByLabelText} = render(
            <MemoryRouter>
                <UserProvider>
                    <TagsSelect selectedTags={mockSelectedTags}
                                handleTagChange={mockHandleTagChange}/>
                </UserProvider>
            </MemoryRouter>);
 
        const Foodie = getByLabelText("Foodie")
        const Explorer = getByLabelText("Explorer")

        // expect already selected tags to be checked in the form
        expect(Foodie).toBeChecked();

        // simulate checking another tag
        fireEvent.click(Explorer);
        // check if handler function works on click
        expect(mockHandleTagChange).toHaveBeenCalled();
    });
});
