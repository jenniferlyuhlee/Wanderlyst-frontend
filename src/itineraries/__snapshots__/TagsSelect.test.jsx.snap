// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TagsSelect sub-form component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Form-tag-check row mt-2"
  >
    <h5
      class="mb-1 p-0"
    >
      Tags:
    </h5>
    <div
      class="form-check"
    >
      <input
        checked=""
        class="form-check-input col"
        id="Foodie"
        type="checkbox"
        value="1"
      />
      <label
        class="form-check-label"
        for="Foodie"
      >
        Foodie
      </label>
    </div>
    <div
      class="form-check"
    >
      <input
        class="form-check-input col"
        id="Explorer"
        type="checkbox"
        value="2"
      />
      <label
        class="form-check-label"
        for="Explorer"
      >
        Explorer
      </label>
    </div>
  </div>
</DocumentFragment>
`;
